### PR TITLE
refactor pilot-agent

### PR DESCRIPTION
**Pull request by knightXun**
refactor pilot-agent

### CODE REVIEW

File: pilot/cmd/pilot-agent/main.go [hunk 1]
```diff
@@ -157,7 +158,7 @@ var (
 			case meshconfig.AuthenticationPolicy_NONE.String():
 				proxyConfig.ControlPlaneAuthPolicy = meshconfig.AuthenticationPolicy_NONE
 			case meshconfig.AuthenticationPolicy_MUTUAL_TLS.String():
-				var ns string
+				var podNamespace string
 				proxyConfig.ControlPlaneAuthPolicy = meshconfig.AuthenticationPolicy_MUTUAL_TLS
```
Comment (qinghai5060) [inline pilot/cmd/pilot-agent/main.go]: no need to modify?

File: pilot/cmd/pilot-agent/main.go [hunk 2]
```diff
@@ -293,7 +294,11 @@ func parseApplicationPorts() ([]uint16, error) {
 			if err != nil {
 				return nil, err
 			}
-			parsedPorts = append(parsedPorts, uint16(parsedPort))
+			if parsedPort > 0 {
+				parsedPorts = append(parsedPorts, uint16(parsedPort))
+			} else {
+				return nil, errors.New("Port must be larger then 0")
+			}
```
Comment (qinghai5060) [inline pilot/cmd/pilot-agent/main.go]: should be "greater then"

File: pilot/cmd/pilot-agent/main.go [hunk 3]
```diff
@@ -32,6 +32,7 @@ import (
 
 	"strconv"
 
+	"github.com/pkg/errors"
```
[the remark this answers is not in the document]
Comment (knightXun) [reply pilot/cmd/pilot-agent/main.go]: that's not a problem

[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pilot/cmd/pilot-agent/main.go]: And what's this pkg? Do you mean errors?

[the remark this answers is not in the document]
Comment (knightXun) [reply pilot/cmd/pilot-agent/main.go]: yes, use it  return an error.

**Comment by knightXun** [comment]
nicely ping@hzxuzhonghu

**Comment by hzxuzhonghu** [comment]
actually, I don't feel strongly about this.

File: pilot/cmd/pilot-agent/main.go [hunk 4]
```diff
@@ -18,6 +18,7 @@ import (
 	"bytes"
 	"context"
 	"encoding/base64"
+	"errors"
```
Comment (hzxuzhonghu) [inline pilot/cmd/pilot-agent/main.go]: I'd rather use `fmt.Errorf` instead of `errors`

**Pull request closed without merging**